test/e2e/httpproxy/include_exact_condition_test.go

```diff
@@ -0,0 +1,215 @@
+						Conditions: []contourv1.MatchCondition{
+							{
+								Exact: "/foo",
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-app",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								Prefix: "/v1",
+							},
+						},
+					},
+				},
+			},
+		}
+		// appProxy will be orphaned when created so can't wait for
+		// it to be valid.
+		require.NoError(t, f.Client.Create(context.TODO(), appProxy))
+
+		adminProxy := &contourv1.HTTPProxy{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: adminNamespace,
+				Name:      "echo-admin",
+			},
+			Spec: contourv1.HTTPProxySpec{
+				Routes: []contourv1.Route{
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-admin",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								Prefix: "/",
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-admin",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								Exact: "/portal",
+							},
+						},
+					},
+				},
+			},
+		}
+		// adminProxy will be orphaned when created so can't wait for
+		// it to be valid.
+		require.NoError(t, f.Client.Create(context.TODO(), adminProxy))
+
+		baseProxy := &contourv1.HTTPProxy{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "echo",
+			},
+			Spec: contourv1.HTTPProxySpec{
+				VirtualHost: &contourv1.VirtualHost{
+					Fqdn: "includeexactcondition.projectcontour.io",
+				},
+				Includes: []contourv1.Include{
+					{
+						Name:      appProxy.Name,
+						Namespace: appProxy.Namespace,
+						Conditions: []contourv1.MatchCondition{
+							{
+								Prefix: "/app",
+							},
+						},
+					},
+					{
+						Name:      adminProxy.Name,
+						Namespace: adminProxy.Namespace,
+						Conditions: []contourv1.MatchCondition{
+							{
+								Prefix: "/app/",
+							},
+						},
+					},
+					{
+						Name:      adminProxy.Name,
+						Namespace: adminProxy.Namespace,
+						Conditions: []contourv1.MatchCondition{
+							{
+								Prefix: "/admin",
+							},
+						},
+					},
+				},
+			},
+		}
+		invalidRootProxy := &contourv1.HTTPProxy{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "echo-invalid",
+			},
+			Spec: contourv1.HTTPProxySpec{
+				VirtualHost: &contourv1.VirtualHost{
+					Fqdn: "includeexactcondition-invalid.projectcontour.io",
+				},
+				Includes: []contourv1.Include{
+					{
+						Name:      appProxy.Name,
+						Namespace: appProxy.Namespace,
+						Conditions: []contourv1.MatchCondition{
+							{
+								Exact: "/app",
+							},
+						},
+					},
+				},
+			},
+		}
+		f.CreateHTTPProxyAndWaitFor(baseProxy, e2e.HTTPProxyValid)
+		f.CreateHTTPProxyAndWaitFor(invalidRootProxy, e2e.HTTPProxyInvalid)
+
+		cases := map[string]string{
```

Comment:
similar nit to the above on comments/naming to make these test cases super clear (looks correct though)